src/ir/LocalGraph.cpp

```diff
@@ -103,12 +99,45 @@ struct Flower : public CFGWalker<Flower, Visitor<Flower>, Info> {
   }
 
   void flow(Function* func) {
+    // Minimalist block information needed
+    struct FlowBlock {
+      size_t lastTraversedIteration; // last Traversed Iteration
+      std::vector<Action> actions; // actions occurring in this block
+      std::vector<FlowBlock*> in;
+      std::vector<std::pair<Index, SetLocal*>> lastSets; // for each index, the last set_local for it
+    };
+
     auto numLocals = func->getNumLocals();
     std::vector<std::vector<GetLocal*>> allGets;
     allGets.resize(numLocals);
-    std::unordered_set<BasicBlock*> seen;
-    std::vector<BasicBlock*> work;
-    for (auto& block : basicBlocks) {
+    std::vector<FlowBlock*> work;
+
+    std::vector<FlowBlock> optimizedBlocks;
+    optimizedBlocks.resize(basicBlocks.size());
+
+    FlowBlock* entryFlowBlock = nullptr;
```

Comment:
how about a comment here like "convert the basicBlocks into more efficient flowBlocks for the flow computation".